src/ModelObjects/representations.rs

```diff
@@ -1,27 +1,31 @@
 use crate::DBMLib::dbm::Zone;
+use crate::ModelObjects::statepair::StatePair;
+use boolean_expression::CubeVar::False;
 use colored::Colorize;
+use generic_array::arr_impl;
 use serde::Deserialize;
+use std::borrow::BorrowMut;
 use std::collections::HashMap;
 use std::fmt::{Display, Formatter};
 use std::ops;
+//use serde::de::Unexpected::Option;
+use serde_json::Value;
+
 /// This file contains the nested enums used to represent systems on each side of refinement as well as all guards, updates etc
 /// note that the enum contains a box (pointer) to an object as they can only hold pointers to data on the heap
 
-#[derive(Debug, Clone, Deserialize, std::cmp::PartialEq, std::cmp::Eq)]
+#[derive(Debug, Clone, Deserialize, PartialEq, Eq)]
 pub enum BoolExpression {
+    Parentheses(Box<BoolExpression>),
     AndOp(Box<BoolExpression>, Box<BoolExpression>),
     OrOp(Box<BoolExpression>, Box<BoolExpression>),
-    LessEQ(Box<BoolExpression>, Box<BoolExpression>),
-    GreatEQ(Box<BoolExpression>, Box<BoolExpression>),
-    LessT(Box<BoolExpression>, Box<BoolExpression>),
-    GreatT(Box<BoolExpression>, Box<BoolExpression>),
-    EQ(Box<BoolExpression>, Box<BoolExpression>),
-    Parentheses(Box<BoolExpression>),
-    Difference(Box<BoolExpression>, Box<BoolExpression>),
-    Clock(u32),
-    VarName(String),
+    LessEQ(Box<ArithExpression>, Box<ArithExpression>),
+    GreatEQ(Box<ArithExpression>, Box<ArithExpression>),
+    LessT(Box<ArithExpression>, Box<ArithExpression>),
+    GreatT(Box<ArithExpression>, Box<ArithExpression>),
+    EQ(Box<ArithExpression>, Box<ArithExpression>),
     Bool(bool),
-    Int(i32),
+    Arithmetic(Box<ArithExpression>),
```

Comment:
I'm unsure what this should be used for? How does it evaluate to a boolean value?

Reply:
This was made for parsing, as parsing a `term` would require returning an  `ArithExpression`. I thought about returning a Trait Object, but we would have to downcast to the concrete type eventually (perhaps using [`Any`](https://doc.rust-lang.org/std/any/trait.Any.html)). I omitted doing this initially because I was focused on producing and solving the task (bad excuse, I know), however I can re-implement it to remove the `Arithmetic` field on `BoolExpression` now if it is desired?

Reply:
It's fine for now, was just interested. We can add a refactor issue for it after this is merged into main